toolbox/relnotes/main.go

```diff
@@ -644,15 +644,30 @@ func patchRelease(f *os.File, info *ReleaseInfo) {
 }
 
 // extractReleaseNoteFromPR tries to fetch release note from PR body, otherwise uses PR title.
-func extractReleaseNoteFromPR(pr *github.Issue) string {
+func extractReleaseNoteFromPR(pr *github.Issue, actionRequired bool) string {
+	var releaseNote string
 	// Regexp Example:
 	// This regexp matches the release note section in Kubernetes pull request template:
 	// https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-	re, _ := regexp.Compile("```release-note\r\n(.+)\r\n```")
+	re := regexp.MustCompile("(?s)```release-note\r\n(.+)\r\n```")
 	if note := re.FindStringSubmatch(*pr.Body); note != nil {
-		return note[1]
+		releaseNote = strings.TrimSpace(note[1])
+	} else {
+		// Use PR title if unable to find release note
+		releaseNote = strings.TrimSpace(*pr.Title)
+	}
+	// We do this for action required PRs only, to avoid removing "action required" message from
+	// some PR that accidentally gets marked as non-action-required PR.
+	if actionRequired {
+		// Formatting the release note. Feel free to add more entries.
+		releaseNote = strings.TrimPrefix(releaseNote, "action required: ")
+		releaseNote = strings.TrimPrefix(releaseNote, "Action required: ")
+		releaseNote = strings.TrimPrefix(releaseNote, "action required:")
+		releaseNote = strings.TrimPrefix(releaseNote, "Action required:")
+		releaseNote = strings.TrimPrefix(releaseNote, "[action required] ")
```

Comment:
Can all of these static strings be fitted into a regexp?

Reply:
Updated. Sorry about that.